File: test-utils/src/main/java/ru/virgil/testutils/fluent_request/RequestModel.java
package ru.virgil.testutils.fluent_request;

import com.fasterxml.jackson.databind.JavaType;
import lombok.Data;
import org.springframework.mock.web.MockMultipartFile;
import org.springframework.test.web.servlet.ResultMatcher;

import javax.annotation.Nullable;

@Data
class RequestModel {

    private RequestMethod requestMethod;
    private String url;
    @Nullable
    private Object requestBody;
    @Nullable
    private JavaType responseJavaType;
    @Nullable
    private MockMultipartFile mockMultipartFile;
    private ResultMatcher[] resultMatchers;

}

File: tools/src/main/kotlin/ru/virgil/spring_tools/tools/testing/fluent_request/RequestModel.kt
package ru.virgil.spring_tools.tools.testing.fluent_request

import com.fasterxml.jackson.databind.JavaType

import org.springframework.mock.web.MockMultipartFile
import org.springframework.test.web.servlet.ResultMatcher


data class RequestModel(
    var requestMethod: RequestMethod? = null,
    var url: String? = null,
    var requestBody: Any? = null,
    var responseJavaType: JavaType? = null,
    var mockMultipartFile: MockMultipartFile? = null,
    var resultMatchers: List<ResultMatcher> = mutableListOf(),
)
